homeassistant/components/omnilogic/switch.py

```diff
@@ -85,8 +87,8 @@ def __init__(
         )
 
         self._state_key = state_key
-        self._state = None
-        self._last_action = 0
+        self._state = False
```

Comment:
Slight code change here, to adjust the default value

Reply:
The reason here is for type hint. My original thought was to set it as `self._state: bool | None = None`, but after looking more closely I felt there was no need to handle `None`.

Looking again now I wonder if there isn't some confusion in this check: `self._state == OMNILOGIC_SWITCH_OFF`
self._state (boolean) will never be equal to SWITCH_OFF (integer) so maybe there is a bigger underlying discrepency.

Reply:
I agree the code in `is_on` seems off.
It should perhaps be like this:
```diff
-            if self._state == OMNILOGIC_SWITCH_OFF:
+            state_int == OMNILOGIC_SWITCH_OFF:
                 state_int = 0
```
@djtimca could you double check? We should mark this PR as draft in the meantime.